model/patch/github.go

```diff
@@ -0,0 +1,131 @@
+package patch
+
+import (
+	"strings"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	// IntentCollection is the database collection that stores patch intents.
+	IntentCollection = "patch_intent"
+
+	// GithubIntentType represents patch intents created for GitHub.
+	GithubIntentType = "github"
+)
+
+// Intent represents an intent to create a patch build and is processed by an amboy queue.
+type Intent interface {
+	// Insert inserts a patch intent in the database.
+	Insert() error
+
+	// SetProcessed should be called by an amboy queue after creating a patch from an intent.
+	SetProcessed() error
+
+	// IsProcessed returns whether a patch exists for this intent.
+	IsProcessed() bool
+
+	// GetType returns the patch intent, e.g., GithubType.
+	GetType() string
+}
+
+// GithubIntent represents an intent to create a patch build as a result of a
+// PullRequestEvent webhook. These intents are processed asynchronously by an
+// amboy queue.
+type GithubIntent struct {
+	// ID is created by the driver and has no special meaning to the application.
+	ID bson.ObjectId `bson:"_id"`
+
+	// PRNumber is the PR number for the project in GitHub.
+	PRNumber int `bson:"pr_number"`
+
+	// HeadSHA is the base SHA of the patch.
+	HeadSHA string `bson:"head_sha"`
```

Comment:
While Git hashes are currently SHA1 (and only SHA1), a number of commits to Git this year changed it so that more algorithms can be supported in the future. Do we want to take this into account with our naming now and call this 'HeadHash'?

Reply:
done